theia-ide/theia
# [vscode] basic support of activation events

#### What it does
- https://github.com/theia-ide/theia/issues/4199 - provides basic mechanism for activation events:
  - compatibility: if an activation event is not supported yet then an extension will be activated eagerly with `*`
  - support of `onLanguage` activation event
  - support of `onCommand` activation event

#### How to test
- unsupported events: deploy vscode go extension and check that it is deployed eagerly since it has unsupported activation events
- `onLanguage` event: deploy https://github.com/akosyakov/lang-activation-event/blob/master/lang-activation-event-0.0.1.vsix and test that `Hello World` notification appears only when typescript file is opened
- `onCommand` event: deploy https://github.com/akosyakov/command-activation-event/blob/master/command-activation-event-0.0.1.vsix and test that `Hello World` command is available without a plugin activation and that it shows `Hello World` notification on the execution

See how to deploy vscode extension here: https://github.com/theia-ide/theia/wiki/Testing-VS-Code-Extensions

#### Review Checklist



- [x] as an author I have carefully reviewed and thoroughly tested my changes.
- [x] commits are signed-off: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md#sign-your-work
- [x] commit history is rebased on master and contains only meaningful commits and changes (less are usually better)
  - for example use `git pull -r` or `git fetch && git rebase` to pick up changes from the master
- [x] [changelog](https://github.com/theia-ide/theia/blob/master/CHANGELOG.md) is updated
- [x] breaking changes are justified and recorded in the changelog
- [x] ~~each new file has a proper copyright with the current year and the name of contributing entity (individual or company)~~
- [x] ~~new dependendencies are justified and [verified](https://github.com/theia-ide/theia/wiki/Registering-CQs#wip---new-ecd-theia-intellectual-property-clearance-approach-experimental)~~
- [x] ~~copied code is justified and [approved via a CQ](https://github.com/theia-ide/theia/wiki/Registering-CQs#case-3rd-party-project-code-copiedforked-from-another-project-into-eclipse-theia-maintained-by-us)~~
- [x] new code is aligned with [project organization](https://github.com/theia-ide/theia/wiki/Code-Organization) and [coding conventions](https://github.com/theia-ide/theia/wiki/Coding-Guidelines)
- [ ] each review approve is provided with a comment regarding whether functionallity is tested, design or code are aligned with project organization and conventions
  - an approve without comment should be dismissed
  - a reviewer should focus on requirements and design validation first, testing second and checking code alignment as the last
